replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/legacy/LegacyDataConverter.java

```diff
@@ -0,0 +1,248 @@
+import org.codice.ditto.replication.api.data.ReplicationSite;
+import org.codice.ditto.replication.api.data.ReplicatorConfig;
+import org.codice.ditto.replication.api.impl.MetacardHelper;
+import org.codice.ditto.replication.api.impl.data.ReplicatorConfigImpl;
+import org.codice.ditto.replication.api.impl.persistence.ReplicationPersistentStore;
+import org.codice.ditto.replication.api.mcard.ReplicationConfig;
+import org.codice.ditto.replication.api.persistence.ReplicatorConfigManager;
+import org.codice.ditto.replication.api.persistence.SiteManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LegacyDataConverter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(LegacyDataConverter.class);
+
+  private static final int DEFAULT_FAILURE_RETRY_COUNT = 5;
+
+  private final CatalogFramework framework;
+
+  private final FilterBuilder filterBuilder;
+
+  private final MetacardHelper helper;
+
+  private final SiteManager siteManager;
+
+  private final ReplicatorConfigManager newConfigManager;
+
+  private final ReplicationPersistentStore persistentStore;
+
+  private final Security security;
+
+  public LegacyDataConverter(
+      CatalogFramework framework,
+      FilterBuilder filterBuilder,
+      MetacardHelper helper,
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager,
+      ReplicationPersistentStore persistentStore) {
+    this(
+        framework,
+        filterBuilder,
+        helper,
+        siteManager,
+        newConfigManager,
+        persistentStore,
+        Security.getInstance());
+  }
+
+  @VisibleForTesting
+  LegacyDataConverter(
+      CatalogFramework framework,
+      FilterBuilder filterBuilder,
+      MetacardHelper helper,
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager,
+      ReplicationPersistentStore persistentStore,
+      Security security) {
+    this.framework = framework;
+    this.filterBuilder = filterBuilder;
+    this.helper = helper;
+    this.siteManager = siteManager;
+    this.newConfigManager = newConfigManager;
+    this.persistentStore = persistentStore;
+    this.security = security;
+  }
+
+  public void init() {
+    Failsafe.with(
+            new RetryPolicy()
+                .withBackoff(3, 60, TimeUnit.SECONDS)
+                .withMaxDuration(10, TimeUnit.MINUTES))
+        .onFailedAttempt(
+            throwable -> LOGGER.debug("failed to upgrade legacy configs, will retry", throwable))
+        .onFailure(
+            throwable ->
+                LOGGER.debug("failed to upgrade legacy configs, restart to try again.", throwable))
+        .run(this::convertConfigs);
+  }
+
+  private void convertConfigs() {
+    // convert sites first
+    persistentStore.objects(OldSite.class).forEach(this::convert020Site);
+
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  getAllConfigs()
```

Comment:
❓ Looks like `getConfigFromMetacard` can return null, should we filter nulls out here?